src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationKeyVaultOptions.cs

```diff
@@ -34,5 +37,23 @@ public AzureAppConfigurationKeyVaultOptions Register(SecretClient secretClient)
             SecretClients.Add(secretClient);
             return this;
         }
+
+        /// <summary>
+        /// Sets the callback used to resolve key vault references that have no registered <see cref="SecretClient"/>.
+        /// </summary>
+        /// <param name="secretResolver">A callback that maps the <see cref="Uri"/> of the secret to its value.</param>
+        /// <remarks>
+        /// The input to the callback will be a key vault secret <see cref="Uri"/>, for example "https://{vault-name}.vault.azure.net/secrets/{name}/{version}"
```

Comment:
Can we omit the URL example. We don't own the format and it looks like we are documenting exactly what the URL will look like. Also, in different clouds, that url format will change.

Reply:
In that case, we don't need this remark at all. The param summary should be sufficient.

Reply:
Oh, yeah I suppose you're right.

Reply:
I think the example Uri is useful to understand what the input to the callback would look like. Even though we don't own the format, our current implementation depends on the Uri to be in this format. We could change `{vault-name}.vault.azure.net` to `vaultBaseUri`.

Reply:
How does our current implementation rely on that format? If it does, we may have a problem.

Reply:
https://github.com/Azure/AppConfiguration-DotnetProvider/blob/1d2a0cb24f0e193907211be4e6d8df0d0a47606e/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs#L41-L42

Reply:
I see we should start using [KeyVaultSecretIdentifier.Parse](https://docs.microsoft.com/en-us/dotnet/api/azure.security.keyvault.secrets.keyvaultsecretidentifier.parse?view=azure-dotnet-preview). Since this next release will be adding a new API, and bumping a minor version. It will probably be a good candidate release for us to bump to [Azure.Security.KeyVault.Secrets 4.1.0](https://www.nuget.org/packages/Azure.Security.KeyVault.Secrets/4.1.0) to use this API.

We have no need to document the URI format. We shouldn't be aware of it.

Reply:
Sure, I can create a new PR for these changes. 
I'll remove the URL example from summary here.